engenere_partner_sales_info/models/res_partner.py

```diff
@@ -0,0 +1,334 @@
+        config_param = self.env["ir.config_parameter"].sudo()
+        return int(
+            config_param.get_param(
+                "engenere_partner_sales_info.default_analysis_months", 24
+            )
+        )
+
+    def _get_start_date(self, analysis_months):
+        return fields.Date.context_today(self) - relativedelta(months=analysis_months)
+
+    def _group_records_by_partner(self, records):
+        grouped = defaultdict(list)
+        for record in records:
+            grouped[record.partner_id.id].append(record)
+        return grouped
+
+    def _compute_record_stats(self, records, date_extractor, amount_extractor):
+        """Calcula estatísticas básicas de uma lista (ordens ou faturas)."""
+        if not records:
+            return None
+
+        sorted_records = sorted(records, key=lambda r: date_extractor(r))
+        dates = [date_extractor(r) for r in sorted_records]
+        amounts = [amount_extractor(r) for r in sorted_records]
+        count = len(sorted_records)
+        total = sum(amounts)
+        average = total / count if count else 0
+
+        avg_no_outliers = average
+        if count >= 3:
+            mean = average
+            variance = sum((x - mean) ** 2 for x in amounts) / count
+            std_dev = math.sqrt(variance)
+            filtered = [
+                x
+                for x in amounts
+                if (mean - 1.5 * std_dev) <= x <= (mean + 1.5 * std_dev)
+            ]
+            avg_no_outliers = sum(filtered) / len(filtered) if filtered else mean
+
+        avg_time_between = 0.0
+        if count >= 2:
+            total_days = sum((dates[i] - dates[i - 1]).days for i in range(1, count))
+            avg_time_between = total_days / (count - 1)
+
+        today = fields.Date.context_today(self)
+        days_since_last = 0
+        if dates and dates[-1]:
+            days_since_last = (today - dates[-1]).days
+
+        return {
+            "last_record": sorted_records[-1],
+            "last_date": dates[-1] if dates else False,
+            "count": count,
+            "total": total,
+            "average": average,
+            "avg_no_outliers": avg_no_outliers,
+            "avg_time_between": avg_time_between,
+            "days_since_last": days_since_last,
+        }
+
+    # ========== ATUALIZA CAMPOS ==========
+
+    def _update_sales_fields(self, stats):
+        self.last_order_id = stats["last_record"].id if stats else False
+        self.last_order_date = stats["last_date"]
+        self.last_order_status = stats["last_record"].state if stats else False
+        self.order_count = stats["count"] or 0
+        self.total_ordered = stats["total"] or 0
+        self.average_ordered = stats["average"] or 0
+        self.average_ordered_no_discrepancies = stats["avg_no_outliers"] or 0
+        self.average_time_between_orders = stats["avg_time_between"] or 0
+        self.days_since_last_order = stats["days_since_last"] or 0
```

Comment:
aqui será que não é melhor assim?
```suggestion
  def _update_sales_fields(self, stats):
      last_record = stats.get("last_record")
      self.write({
          'last_order_id': last_record.id if last_record else False,
          'last_order_date': stats.get("last_date"),
          'last_order_status': last_record.state if last_record else False,
          'order_count': stats.get("count", 0),
          'total_ordered': stats.get("total", 0),
          'average_ordered': stats.get("average", 0),
          'average_ordered_no_discrepancies': stats.get("avg_no_outliers", 0),
          'average_time_between_orders': stats.get("avg_time_between", 0),
          'days_since_last_order': stats.get("days_since_last", 0),
      })
```